tracer/src/Datadog.Trace/Propagators/PropagationContext.cs

```diff
@@ -13,10 +16,20 @@ internal readonly struct PropagationContext
 
     public readonly Baggage? Baggage;
 
+    public readonly IEnumerable<SpanLink>? Links;
+
     public PropagationContext(SpanContext? spanContext, Baggage? baggage)
     {
         SpanContext = spanContext;
         Baggage = baggage;
+        Links = Enumerable.Empty<SpanLink>();
```

Comment:
Initializing `Links` to non-null in the ctors leads to some inconsistency:
```csharp
// Links is null (ctor not called)
PropagationContext a = default;

// Links is []
PropagationContext b = new(spanContext: null, baggage: null);

// Links is [] even when passing extractionSpanLinks: null
PropagationContext b = new(spanContext: null, baggage: null, extractionSpanLinks: null);
```

Should we let `Links` be `null` in the ctors?

Reply:
I was hoping to avoid nulls entirely, but if `default(PropagationContext)` will leave the `Links` property as null then I think we should allow `Links` to be `null` in the ctors instead of `[]`. Do you agree?

Reply:
I've changed this in 9b8ccc7466fa356c69cad26d1d6a42a2e21a9018 to allow nulls. WDYT?

Reply:
Thanks